.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review, review_requested]
+  issue_comment:
+    types: [created]
+
+jobs:
+  code_review_job:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: write
+    name: Run code review agent on every pull request, respond to user comments
+
+    steps:
+      - name: Set Options Environment Variable for Issue Comment
+        if: github.event_name == 'issue_comment'
+        run: |
```

Comment:
**Performance Issue**: The use of static analysis and code feedback options in the GitHub Actions workflow may significantly increase the execution time of the CI pipeline, especially for larger projects. The static analysis tool, fb_infer, and the comprehensive code feedback mechanism can be resource-intensive, leading to longer wait times for CI checks to complete. <br> **Fix**: Consider conditionally enabling these features based on the type of pull request or the files changed. For instance, you could limit the execution of these tools to pull requests targeting specific branches or only run them when changes to critical parts of the codebase are detected. This approach can help balance the need for thorough code quality checks with the necessity of maintaining a swift CI pipeline. <br> **Code Suggestion**: 
 ```
 if: github.event_name == 'pull_request'
 run: |
   if [ "${{ github.base_ref }}" == "main" ] || [[ "${{ github.files_changed }}" =~ "critical_path/" ]]; then
     OPTIONS="--static_analysis.fb_infer.enabled=True --code_feedback=True"
   else
     OPTIONS="--code_feedback=True"
   fi
   echo "OPTIONS=${OPTIONS} --dependency_check.enabled=False --bee.path=/automation-platform  --bee.actn_dir=/automation-platform/default_bito_ad/bito_modules --git.access_token=${{ secrets.GIT_ACCESS_TOKEN }} --bito_cli.bito.access_key=${{ secrets.BITO_ACCESS_KEY }} ${{ secrets.GIT_DOMAIN && ' --git.domain=' || '' }}${{ secrets.GIT_DOMAIN }}" >> $GITHUB_ENV
 ```